cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/AppLifecycleHttpHandlerTest.java

```diff
@@ -126,6 +127,25 @@ public void testDeployUsingArtifact() throws Exception {
       doDelete(getVersionedAPIPath("apps/" + appId.getId(), appId.getNamespaceId())).getStatusLine().getStatusCode());
   }
 
+  @Test
+  public void testDeployVersionedApp() throws Exception {
```

Comment:
We should add tests for - get versioned app, getting app without versions, deleting